gemd/entity/object/base_object.py

```diff
@@ -54,3 +54,50 @@ def file_links(self):
     @file_links.setter
     def file_links(self, file_links):
         self._file_links = validate_list(file_links, FileLink)
+
+    def all_dependences(self):
+        """Return a set of all immediate dependencies (no recursion)."""
```

Comment:
Why no recursion? Does method you added to citrine_python traverse the whole graph or just adjacent nodes?

Reply:
Since GEMD objects are now comparable by equality, couldn't you just call `.all_dependencies` on each object you find since this method exists on all entity types. And then `set()` the outputs to remove any duplicates?

Reply:
The intent is to only return one layer of dependencies.  This isn't as simple as filtering the values of an as_dict because, for example, attributes may contain templates, and that requires navigating 2 layers of objects.  On the other hand, you don't want to include the spec.template object.